optuna/integration/skorch.py

```diff
@@ -0,0 +1,45 @@
+from typing import Any
+
+import optuna
+
+with optuna._imports.try_import() as _imports:
+    from skorch.callbacks import Callback
+    from skorch.net import NeuralNet
+
+if not _imports.is_successful():
+    Callback = object  # NOQA
+    NeuralNet = object  # NOQA
+
+
+class SkorchPruningCallback(Callback):
+    """Skorch callback to prune unpromising trials.
+
+    Args:
+        trial:
+            A :class:`~optuna.trial.Trial` corresponding to the current evaluation of the
+            objective function.
+        monitor:
+            An evaluation metric for pruning, e.g. ``val_loss`` or
+            ``val_acc``. The metrics are obtained from the returned dictionaries,
+            i.e., ``net.histroy``. The names are thus depend on how this dictionary
+            is formatted.
+    """
+
+    def __init__(self, trial: optuna.trial.Trial, monitor: str) -> None:
+
+        _imports.check()
+
+        super().__init__()
+        self._trial = trial
+        self._monitor = monitor
+
+    def on_epoch_end(self, net: NeuralNet, **kwargs: Any) -> None:
```

Comment:
this should be
```suggestion
    def on_epoch_end(self, net: "skorch.NeuralNet", **kwargs: Any) -> None:
```
?

Reply:
Yes, I agree with you. As @hvy pointed in https://github.com/optuna/optuna/pull/1705#discussion_r476296682, this is rendered in the docs as `object`. Using `"skorch.NeuralNet"` instead `NeuralNet` leads to informative documentation.